core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -250,10 +265,93 @@ public void setConf(Configuration conf) {
     this.conf = conf;
   }
 
+  private boolean insertProperties(Namespace namespace, Map<String, String> properties) {
+    String namespaceName = JdbcUtil.namespaceToString(namespace);
+
+    try {
+      int insertedRecords = connections.run(conn -> {
+        String sqlStatement = JdbcUtil.insertPropertiesStatement(properties);
+
+        try (PreparedStatement sql = conn.prepareStatement(sqlStatement)) {
+          int rowIndex = 0;
+          for (Map.Entry<String, String> keyValue : properties.entrySet()) {
+            sql.setString(++rowIndex, catalogName);
+            sql.setString(++rowIndex, namespaceName);
+            sql.setString(++rowIndex, keyValue.getKey());
+            sql.setString(++rowIndex, keyValue.getValue());
+          }
+          return sql.executeUpdate();
+        }
+      });
+
+      if (insertedRecords == properties.size()) {
+        LOG.debug("Successfully committed to new namespace: {}", namespaceName);
+        return true;
+      } else {
+        throw new CommitFailedException("Failed to insertProperties to namespace %s in catalog %s", namespaceName,
+                catalogName);
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new UncheckedInterruptedException(e,
+              "Interrupted in call to insertProperties(namespace, properties) Namespace: %s", namespace);
+    } catch (SQLException e) {
+      throw new UncheckedSQLException(e, "Failed to insertProperties to namespace: %s in catalog: %s", namespace,
+              catalogName);
+    }
+  }
+
+  private boolean updateProperties(Namespace namespace, Map<String, String> properties) {
```

Comment:
I would personally say if there's no practical difference between the two, it would be ok to make them into one function.

Possibly there could be one outer function, that contains all of the shared logic (like the interrupted exception and SQLException checks). And then two different functions that wrap that function with a code block for the building of the correct prepared statement. This might make the changes a bit easier to digest and be cleaner, but it's fine as it is otherwise.